yoti/tests/src/Unit/YotiHelperTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\Tests\yoti\Unit;
+
+use Drupal\Tests\UnitTestCase;
+use Drupal\yoti\YotiHelper;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Cache\CacheTagsInvalidatorInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Database\Query\Delete;
+
+/**
+ * @coversDefaultClass \Drupal\yoti\YotiHelper
+ *
+ * @group yoti
+ */
+class YotiHelperTest extends UnitTestCase {
+
+  /**
+   * Setup for YotiHelper tests.
+   */
+  public function setUp() {
+    // Setup container to handle situations where dependencies
+    // haven't been injected.
+    $container = new ContainerBuilder();
+    $container->set('config.factory', $this->getMockConfigFactory());
+    $container->set('current_user', $this->getMock(AccountProxyInterface::class));
+    $container->set('messenger', $this->getMock(MessengerInterface::class));
+    $container->set('database', $this->getMockDatabase());
+    \Drupal::setContainer($container);
+  }
+
+  /**
+   * @covers ::link
+   * @backupGlobals enabled
+   */
+  public function testLinkInvalidToken() {
+    // Cache should not be invalidated if invalid token is provided.
+    $cacheTagsInvalidator = $this->getMock(CacheTagsInvalidatorInterface::class);
+    $cacheTagsInvalidator
+      ->expects($this->exactly(0))
+      ->method('invalidateTags');
+
+    $helper = new YotiHelper($this->getEntityTypeManager(), $cacheTagsInvalidator);
+
+    // Attempt link with no token.
+    $result = $helper->link();
+    $this->assertFalse($result);
+
+    // Attempt link with an invalid token.
+    $_GET['token'] = 'test_token';
+    $result = $helper->link();
+    $this->assertFalse($result);
+  }
+
+  /**
+   * @covers ::unlink
+   */
+  public function testUnlinkCacheInvalidation() {
+    // Cache should be invalidated when user is unlinked.
+    $cacheTagsInvalidator = $this->getMock(CacheTagsInvalidatorInterface::class);
+    $cacheTagsInvalidator
+      ->expects($this->once())
+      ->method('invalidateTags')
+      ->with($this->equalTo(['tag:1', 'tag:2']));
+
+    $helper = new YotiHelper($this->getEntityTypeManager(), $cacheTagsInvalidator);
+    $helper->unlink();
+  }
+
+  /**
+   * Get mock for entity type manager.
+   *
+   * @return \Drupal\Core\Entity\EntityTypeManagerInterface
+   *   Mocked entity type manager
+   */
+  private function getEntityTypeManager() {
```

Comment:
There's a lot of *get*XXX methods in here.  Are they actually creating stuff?  Maybe *create*XXX, or even *create*XXX*Mock*?

Reply:
Agree, I'll change these to "create" as it's creating mocks